src/store/emptyImageBuilderApi.ts

```diff
@@ -5,6 +5,32 @@ import { IMAGE_BUILDER_API } from '../constants';
 // initialize an empty api service that we'll inject endpoints into later as needed
 export const emptyImageBuilderApi = createApi({
   reducerPath: 'imageBuilderApi',
-  baseQuery: fetchBaseQuery({ baseUrl: IMAGE_BUILDER_API }),
+  baseQuery: fetchBaseQuery({
+    baseUrl: IMAGE_BUILDER_API,
+    paramsSerializer: (params) => {
+      /*
+       * Image builder backend requires the arrays in get requests to be
+       * exploded, see the default behavior for swagger in the documentation:
+       * https://swagger.io/docs/specification/serialization/
+       *
+       * To accommodate to that, make sure that when the request is sent with
+       * an array we do explode properly unlike the default behavior of
+       * URLSearchParams.
+       */
+      const searchParams = new URLSearchParams();
+      for (const [key, value] of Object.entries(params)) {
+        if (value === undefined) {
+          // avoid sending undefined parameters
+          continue;
+        }
+        if (Array.isArray(value)) {
+          value.forEach((v) => searchParams.append(key, v));
+        } else {
+          searchParams.append(key, value);
+        }
+      }
+      return searchParams.toString();
+    },
+  }),
```

Comment:
@lucasgarfield and @regexowl I really want your opinion on this chunk. I think it's all good now, but it's only handling arrays not objects. I don't know if we should support object as get parameters, I don't think it makes much sense tbh. It would also be possible that this breaks in the future. I'm slightly worried :sweat_smile: